utilities/node_activator/test/node_activator_test.exs

```diff
@@ -2,13 +2,64 @@ defmodule NodeActivatorTest do
   use ExUnit.Case
   doctest NodeActivator
 
-  test "run" do
-    NodeActivator.run("test")
+  setup do
+    on_exit(fn -> Node.stop() end)
+  end
+
+  describe "run" do
+    test "starts node with correct name" do
+      refute Node.alive?()
+
+      {:ok, node_name} = NodeActivator.run("test prefix")
+
+      assert Node.alive?()
+      assert Node.self() == node_name
+      assert "test-prefix" <> <<_::binary>> = to_string(node_name)
+    end
+
+    test "is idempotent" do
+      {:ok, node_name1} = NodeActivator.run("test prefix")
+      {:ok, node_name2} = NodeActivator.run("test prefix")
+
+      assert node_name1 == node_name2
+    end
+  end
+
+  describe "generate_node_name" do
+    test "generates unique atom" do
+      node_name1 = NodeActivator.generate_node_name("test prefix")
+      node_name2 = NodeActivator.generate_node_name("test prefix")
+
+      assert is_atom(node_name1)
+      refute node_name1 == node_name2
+    end
+
+    test "starts with provided prefix" do
+      node_name = NodeActivator.generate_node_name("test prefix")
+
+      assert node_name |> Atom.to_string() |> String.starts_with?("test-prefix_")
+    end
+
+    test "ends with hostname" do
+      node_name = NodeActivator.generate_node_name("test prefix")
+
+      {:ok, hostname} = :inet.gethostname()
```

Comment:
On macOS, `:inet.gethostname()` returns `~c"hostname"`, but the long name of Node is `"hostname.local"`, so they does not match.

`:inet.gethostname()` should be replaced by `NodeActivator.get_hostname`.